front/tests/008-allowances.spec.ts

```diff
@@ -25,41 +19,30 @@ test.describe('Testing if all mandatory elements simulation configuration are lo
 
     await expect(scenarioPage.getTimetableList).toBeVisible();
 
-    await scenarioPage.clickBtnByName('Afficher/masquer le détail des trains');
-
-    await scenarioPage.getBtnByName(/Train Schedule 1/).hover();
-
-    await scenarioPage.clickBtnByName('Modifier');
+    await scenarioPage.getBtnByName(scenarioName).hover();
+    await page.getByTestId('edit-train').click();
 
+    // Add and check standard allowance
     await scenarioPage.openAllowancesModule();
     await expect(scenarioPage.getAllowancesModule).toBeVisible();
-
-    // Add and check standard allowance
     await scenarioPage.setStandardAllowance();
-    await scenarioPage.clickBtnByName('Modifier le train');
-
+    await page.getByTestId('submit-edit-train-schedule').click();
     expect(await scenarioPage.isAllowanceWorking()).toEqual(true);
 
-    await scenarioPage.getBtnByName(/Train Schedule 1/).hover();
-    await scenarioPage.clickBtnByName('Modifier');
-    await scenarioPage.openAllowancesModule();
-    await expect(scenarioPage.getAllowancesModule).toBeVisible();
+    await scenarioPage.getBtnByName(scenarioName).hover();
+    await page.getByTestId('edit-train').click();
 
     // Add and check engineering allowance
+    await scenarioPage.openAllowancesModule();
+    await expect(scenarioPage.getAllowancesModule).toBeVisible();
     await scenarioPage.setEngineeringAllowance();
     await scenarioPage.clickSuccessBtn();
     expect(scenarioPage.checkAllowanceEngineeringBtn());
-    await scenarioPage.clickBtnByName('Modifier le train');
+    await page.getByTestId('submit-edit-train-schedule').click();
     await scenarioPage.page.waitForSelector('.scenario-details-name');
 
     expect(await scenarioPage.isAllowanceWorking()).toEqual(true);
```

Comment:
As we are adding a second margin, we don't check if this engineering margin changed anything at all. 

I think it would be better to extract the first allowancesEnergyConsumed and check that the new one with the engineering is smaller than this (or remove the standard allowance).

Reply:
in this case it should be better for each allowance type to have his own test in the file

Reply:
done